staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -298,13 +322,28 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
-	serveFunc()
+	serve := func() {
+		requestInfo, _ := apirequest.RequestInfoFrom(req.Context())
+		metrics.RecordLongRunning(req, requestInfo, metrics.APIServerComponent, func() {
```

Comment:
Same here - moving this part to that separate PR I mentioned above would make it easier to reason about everything.